specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -0,0 +1,665 @@
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ComputeDiagnosticBase"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the get call failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the status of a scheduled disk inspection request.": {
+            "$ref": "./examples/GetDiskInspection.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/registerStorageConfiguration": {
+      "post": {
+        "tags": [
+          "registerStorageConfiguration"
+        ],
+        "operationId": "DiskInspectionStorageConfiguration_Register",
+        "x-ms-long-running-operation": false,
+        "description": "Register a storageAccount for a subscription used for DiskInspection",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "storageConfigurationInput",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/StorageConfigurationInput"
+            },
+            "description": "Storage Configuration object supplied in the body of the Post to cache storageAccount for a given subscription."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Succeeded"
+          },
+          "default": {
+            "description": "Error response describing why the register operation failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create a request for registering a storageAccount information.": {
+            "$ref": "./examples/RegisterStorageConfiguration.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/validateStorageConfiguration": {
```

Comment:
>diskInspection/validateStorageConfiguration" [](http://example.com/codeflow?start=98&length=44)

this would be breaking, but in retrospect, I wonder, why it wasn't e.g. "diagnostics/validateDiskStorageConfiguration/run" - I don't really understand the distinctions being made by having a group under diskInspections

Reply:
Resolving due to earlier conversation about resourcetype/action design